pkg/tfconf/resource.go

```diff
@@ -0,0 +1,40 @@
+package tfconf
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/terraform-docs/terraform-docs/internal/types"
+)
+
+// Resource represents a managed or data type that is created by the module
+type Resource struct {
+	Type           string       `json:"type" xml:"type" yaml:"type"`
+	ProviderName   string       `json:"providerName" toml:"providerName" xml:"providerName" yaml:"providerName"`
+	ProviderSource string       `json:"provicerSource" toml:"providerSource" xml:"providerSource" yaml:"providerSource"`
+	Mode           string       `json:"mode" tomml:"mode" xml:"mode" yaml:"mode"`
+	Version        types.String `json:"version" xml:"version" yaml:"version"`
+}
+
+// FullType returns full name of the type of the resource, including the provider name
+func (r *Resource) FullType() string {
+	return r.ProviderName + "_" + r.Type
+}
+
+// URL returns a best guess at the URL for resource documentation
+func (r *Resource) URL() string {
+	kind := ""
+	switch r.Mode {
+	case "managed":
+		kind = "resources"
+	case "data":
+		kind = "data-sources"
+	default:
+		return ""
+	}
+
+	if strings.Count(r.ProviderSource, "/") > 1 {
```

Comment:
I'm curious under which circumstance this happens, or is it just a safeguard to prevent any malformed data?

Reply:
This intended to catch private/custom provider sources. Unless I'm mistaken, the custom/private providers would be provided as a fully qualified URL instead of "hashicorp/foo".

Reply:
I'm not entirely familiar with the inner mechanic of provider registry, but I did a quick look around and what I gather is that you, as a provider owner, either publish your provider to registry.terraform.io or not (not entirely sure what the alternative is though. A private registry?).

That being said, I think I'm fine with this condition as is (at least for now, unless we gather more information) because even if the `source` has more than 1 `/` (i.e. a full URL of a registry and a provider) there still won't be any predictable way to construct the URL. But in this case the generated output (for example in markdown table) won't be correct and produces a warning `- [myprovider_someresource]()`. Maybe we need to consider this when generating the content with go template.

Reply:
Good call. I'll take another look at this and prevent those invalid links from being generated.

Reply:
Awesome, thanks @xortim :+1: